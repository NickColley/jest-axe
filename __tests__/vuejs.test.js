const { mount } = require("@vue/test-utils");
const { render } = require("@testing-library/vue");

const { axe, toHaveNoViolations } = require("../index");

const Image = {
  data: () => ({ src: "#" }),
  template: '<img id="test-image" :src="src" />',
};

expect.extend(toHaveNoViolations);

describe("Vue", () => {
  it("renders correctly", async () => {
    const wrapper = mount(Image);
    const results = await axe(wrapper.element);

    expect(() => {
      expect(results).toHaveNoViolations();
    }).toThrowErrorMatchingSnapshot();
  });

  it("renders a vue testing library container correctly", async () => {
    const { container } = render(Image);
    const results = await axe(container);

    expect(() => {
      expect(results).toHaveNoViolations();
    }).toThrowErrorMatchingSnapshot();
  });
});
