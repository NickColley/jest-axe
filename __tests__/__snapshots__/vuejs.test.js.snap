// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vue renders a vue testing library container correctly 1`] = `
"expect(received).toHaveNoViolations(expected)

Expected the HTML found at $('#test-image') to have no violations:

<img id="test-image" src="#">

Received:

"Images must have alternate text (image-alt)"

Fix any of the following:
  Element does not have an alt attribute
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element has no title attribute
  Element's default semantics were not overridden with role="none" or role="presentation"

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.5/image-alt?application=axeAPI"
`;

exports[`Vue renders correctly 1`] = `
"expect(received).toHaveNoViolations(expected)

Expected the HTML found at $('#test-image') to have no violations:

<img id="test-image" src="#">

Received:

"Images must have alternate text (image-alt)"

Fix any of the following:
  Element does not have an alt attribute
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element has no title attribute
  Element's default semantics were not overridden with role="none" or role="presentation"

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.5/image-alt?application=axeAPI

────────

Expected the HTML found at $('#test-image') to have no violations:

<img id="test-image" src="#">

Received:

"All page content should be contained by landmarks (region)"

Fix any of the following:
  Some page content is not contained by landmarks

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.5/region?application=axeAPI"
`;
